### ISSUE TITLE
only auto save working copy when content has changed since last auto save

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#3934]
When the html file is dirty we auto save the working copy to the server every 30 seconds or so. even if the content has not changed since the last auto save. We should only put the working copy of the file to the server if the content has change since the last auto save.

**Comment by JonFerraiolo** [comment]
@billreed63 - are you sure that we auto-save every 30 seconds even if the file is unchanged? At one point I was chasing through the code and noticed that we check every 30 seconds, but only PUT the file if the file has been marked dirty.

**Comment by billreed63** [comment]
exactly, if the file is marked dirty we auto save, then I make no more changes to the file.. then after 30 seconds we auto save. the file is still dirty but the working copy is unchanged since the last auto save. we do another put of the working copy even though it is not necessary because the  working copy on the server is at the save level as the client version.

**Comment by peller** [comment]
upon quick inspection, it looks like there are at least 3 different dirty bits: one on the model, one on the page editor, one on the visual editor.  dirty is set to true for an autosave; it apparently means a "real" or user-initiated save has to happen, as opposed to a save which does a put to save a working copy.  The lack of clarity is probably responsible for most of our dirty save bugs.  Not something we should try to touch for M10.
